src/cheri_ptw.sail

```diff
@@ -24,22 +24,62 @@ function ptw_error_to_str(e) =
 
 overload to_str = {ptw_error_to_str}
 
+/*
+ * Produce the PTW_Error for PTE_Check_Failure-s arising from
+ * checkPTEPermission calls.  This function is not invoked for
+ * PTE_Check_Success values nor on any other ext_ptw values; this explains the
+ * certainty expressed with our PTW_Error return type.  Our result always makes
+ * its way to translationException, below, so some conspiracy is possible: even
+ * though we cannot see the ext_access_type for the transaction at hand, and we
+ * did not carry it through from where it was visible to checkPTEPermission, it
+ * is again visible to translationException.
+ *
+ * We analyse only the store-side component of ext_ptw here.  The MMU/PTW logic
+ * does not see into the data path, and so load-side responses, including
+ * traps, must be handled by the instruction that initiated the translation by
+ * considering the ext_ptw result provided by the MMU/PTW.
+ */
```

Comment:
Well, sure, not here, I guess, but it would have been incredibly helpful if there had been a kind of "information flow overview" of the PTW logic, as it's spread across multiple files and partially overridden between this and the baseline model.